.github/actions/wsl-install/action.yaml

```diff
@@ -0,0 +1,50 @@
+name: Install Ubuntu WSL
+description: 'Install WSL from the store'
+
+inputs:
+  distro:
+    description: Distro name as shown in 'wsl -l -v'. Leave blank to install no distro.
+    required: false
+    default: ""
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Install WSL
+      shell: powershell
+      run: |
+        # Install or update WSL
+        $env:WSL_UTF8=1
+        winget install --name 'Windows Subsystem for Linux' --accept-source-agreements --accept-package-agreements --silent
+        if ( ! $? ) { Exit(1) }
+
+    - name: Install the new distro
+      if: ${{ inputs.distro != '' }}
+      uses: ./.github/actions/wsl-powershell
+      with:
+        exec: |
+          # Install the new distro
+          switch ( "${{ inputs.distro }}" )
+          {
+              "Ubuntu"         { $storeId="9PDXGNCFSCZV" ; $launcher="ubuntu.exe"         ;  $appxname="Ubuntu"          }
+              "Ubuntu-Preview" { $storeId="9P7BDVKVNXZ6" ; $launcher="ubuntupreview.exe"  ;  $appxname="UbuntuPreview"   }
+              "Ubuntu-22.04"   { $storeId="9PN20MSR04DW" ; $launcher="ubuntu2204.exe"     ;  $appxname="Ubuntu22.04LTS"  }
+              "Ubuntu-20.04"   { $storeId="9MTTCL66CPXJ" ; $launcher="ubuntu2004.exe"     ;  $appxname="Ubuntu20.04LTS"  }
+              "Ubuntu-18.04"   { $storeId="9PNKSF5ZN4SW" ; $launcher="ubuntu1804.exe"     ;  $appxname="Ubuntu18.04LTS"  }
+              default          { Write-Output "Error: Unknown distro ${distro}" ; Exit(1) }
+          }
+
+          # Unregister and uninstall pre-existing distro
+          Remove-AppxPackage "$(Get-AppxPackage | Where-Object Name -like "CanonicalGroupLimited.${appxname}")" 3>&1 2>&1 | Out-Null
```

Comment:
You are using like here, IIRC, you told that it was replaced with "foo*", so it means that if we set the name to `Ubuntu`, can this may be transformed into `-like "CanonicalGroupLimited.Ubuntu*"` and thus match any other distro?

Reply:
The opposite. It used to be `*distro-name*`, now it is an exact match. I could change it to 
```diff
- Where-Object Name -like "CanonicalGroupLimited.${appxname}"
+ Where-Object Name -value "CanonicalGroupLimited.${appxname}"
```
but the result is the same (-like and -value do the same thing if there are no asterisks).

Reply:
Ok, I was afraid of the unexpected regexp behaviour completing the end :)